### PR TITLE
Add SSH proxy support

### PR DESCRIPTION
**Pull request by stengaard**
A lot of setups tunnel through a jump host (bastion host), this adds support for tunnelling in `sup`.

### CODE REVIEW

File: ssh.go [hunk 1]
```diff
@@ -108,21 +108,33 @@ func initAuthMethod() {
...
 func (c *SSHClient) Connect(host string) error {
+	return c.ConnectWith(host, ssh.Dial)
+}
+
+// ConnectWith creates a SSH connection to a specified host. It will use dialer to establish the
+// connection.
+// TODO: Split Signers to its own method.
+func (c *SSHClient) ConnectWith(host string, dialer SSHDialFunc) error {
+
+	fmt.Println("connecting", host)
```
Comment (pkieltyka) [inline ssh.go]: Thanks for this PR- awesome stuff!

can you remove this debug Println. Or we should have a sup verbose mode, and have a logger there, etc. For now, remove this line.

File: ssh.go [hunk 2]
```diff
@@ -195,6 +208,21 @@ func (c *SSHClient) Wait() error {
 	return err
 }
 
+// DialThrough will create a new connection from the ssh server sc is connected to. DialThrough is an SSHDialer.
+func (sc *SSHClient) DialThrough(net, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
+	fmt.Println("Dialing", net, addr)
```
Comment (pkieltyka) [inline ssh.go]: remove

File: supfile.go [hunk 3]
```diff
@@ -18,15 +18,16 @@ type Supfile struct {
...
-	Env   map[string]string `yaml:"env"`
+	Bastion string            `yaml:"bastion"` // Jump host for the environment
+	Hosts   []string          `yaml:"hosts"`
+	Env     map[string]string `yaml:"env"`
 }
 
 // Command represents command(s) to be run remotely.
 type Command struct {
-	Name    string   `yaml:-`          // Command name.
+	Name    string   `yaml:"-"`        // Command name.
```
Comment (pkieltyka) [inline supfile.go]: Good catch.

File: ssh.go (hunk 1 above)
Comment (stengaard) [inline ssh.go]: Ow shit - yes. I'll drop those. Thanks.

**Comment by VojtechVitek** [comment]
Very cool addition. Thanks!

**Pull request merged**